compiler/modules/nback/structure.bal

```diff
@@ -83,21 +83,35 @@ function buildListConstruct(llvm:Builder builder, Scaffold scaffold, bir:ListCon
 
         // Cases that are not UniformTypeBitSet should have been filtered out before
         t:UniformTypeBitSet memberType = <t:UniformTypeBitSet>t:simpleArrayMemberType(scaffold.typeContext(), listType);
-        foreach int i in 0 ..< length {
-            builder.store(check buildWideRepr(builder, scaffold, insn.operands[i], REPR_ANY, memberType),
-                          builder.getElementPtr(array, [llvm:constInt(LLVM_INT, 0), llvm:constInt(LLVM_INT, i)], "inbounds"));
+        // pr-todo: should I replace if-else with a lookup table?
+        if memberType == t:INT {
+            check buildListArray(builder, scaffold, array, LLVM_INT, insn.operands, o => buildInt(builder, scaffold, <bir:IntOperand>o));
```

Comment:
I don't think we have any closures in the code base at the moment: I have been avoiding them to facilitate self-hosting.

Reply:
The function argument is duplicating the information in the fourth argument.